src/state/profile/index.tsx

```diff
@@ -16,10 +17,13 @@ export const profileSlice = createSlice({
     profileFetchStart: (state) => {
       state.isLoading = true
     },
-    profileFetchSucceeded: (state, action: PayloadAction<Profile>) => {
+    profileFetchSucceeded: (state, action: PayloadAction<GetProfileResponse>) => {
+      const { profile, hasRegistered } = action.payload
+
       state.isInitialized = true
       state.isLoading = false
-      state.data = action.payload
+      state.hasRegistered = hasRegistered
```

Comment:
fyi I think you can use 
```
return {
isInitialized: ...
....
}
```

if you want to update the whole root state object

Reply:
I'm using this here https://github.com/pancakeswap/pancake-frontend/pull/381/files#diff-ca66a16ecdb113ac49e16d228ee03bad184f6ca83318f18e5993831a31588250R21

Reply:
Ah, nice to know! I always thought it was cumbersome updating each variable.